hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -591,6 +720,31 @@ int readRemote(long position, byte[] b, int offset, int length, TracingContext t
     return (int) bytesRead;
   }
 
+  private void initPropertiesFromReadResponseHeader(final AbfsHttpOperation op) throws IOException {
+    if (DIRECTORY.equals(
```

Comment:
Its duplicated in two places. Probably you can create a method and use it


```
    if (DIRECTORY.equals(
        op.getResponseHeader(HttpHeaderConfigurations.X_MS_RESOURCE_TYPE))) {
      throw new FileNotFoundException(
          "read must be used with files and not directories. Path: " + path);
    }
    contentLength = parseFromRange(
        op.getResponseHeader(HttpHeaderConfigurations.CONTENT_RANGE));
    eTag = op.getResponseHeader(HttpHeaderConfigurations.ETAG);
    if (eTag != null && contentLength >= 0) {
      fileStatusInformationPresent = true;
    }
  }
```

Reply:
Makes sense. Have taken out method `validateFileResourceTypeAndParseETag` that checks the resourceType and parses Etag. Can be used both places.